lib/analytics_creator.rb

```diff
@@ -0,0 +1,66 @@
+class AnalyticsCreator
+  SCHEMA = "public"
+  attr_reader :username, :password, :database_name
+
+
+  def initialize(username:, password:, database_name: nil)
+    @username = username
+    @password = password
+    @database_name = database_name || ActiveRecord::Base.connection_db_config.database
+  end
+
+  def run
+    create_user
+    apply_permissions
+  end
+
+  def drop_user
+    roles = execute("select * from pg_catalog.pg_roles where rolname='#{username}'")
+    return unless roles.rows.any?
+
+    execute("REVOKE ALL PRIVILEGES ON ALL TABLES IN SCHEMA #{SCHEMA} FROM #{username};")
+    execute("REVOKE USAGE ON SCHEMA #{SCHEMA} FROM #{username}")
+    execute("ALTER DEFAULT PRIVILEGES IN SCHEMA #{SCHEMA} REVOKE ALL ON TABLES FROM #{username}")
+    execute("REVOKE CONNECT ON DATABASE #{database_name} FROM #{username}")
+    execute("DROP ROLE #{username}")
+  end
+
+private
+
+  def create_user
+    raise "ANALYTICS_DB_PASSWORD must be set." if password.blank?
+
+    roles = execute("select * from pg_catalog.pg_roles where rolname='#{username}'")
+    return if roles.rows.any?
+
+    execute("CREATE ROLE #{username} WITH LOGIN PASSWORD '#{password}'")
+    execute("GRANT CONNECT ON DATABASE #{database_name} TO #{username}")
+    execute("GRANT USAGE ON SCHEMA #{SCHEMA} TO #{username}")
```

Comment:
should we not just be "grant select"ing - least privilege principl. [USAGE](https://www.postgresql.org/docs/9.1/sql-grant.html) may grant more than we need or not enough to select from views?!

Reply:
Usage is need to access the schema... :) without it the account can't do anything based off:

https://stackoverflow.com/questions/17338621/what-does-grant-usage-on-schema-do-exactly